cmd/oras/manifest/fetch.go

```diff
@@ -87,17 +93,18 @@ Example - Fetch manifest with prettified json result:
 func fetchManifest(opts fetchOptions) (fetchErr error) {
 	ctx, _ := opts.SetLoggerLevel()
 
-	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
+	target, err := opts.NewReadonlyTarget(ctx, opts.Common)
 	if err != nil {
 		return err
 	}
-
-	if repo.Reference.Reference == "" {
-		return oerrors.NewErrInvalidReference(repo.Reference)
+	if opts.Reference == "" {
+		return oerrors.NewErrInvalidReferenceStr(opts.Reference)
```

Comment:
Just a question, what do the `opts.Reference` and `opts.RawReference` look like?

Reply:
`opts.RawReference` is the full user input, e.g. `localhost:5000/test@sha256:xxxx`. `opts.Reference` is the tag or digest part, e.g. `sha256:XXXX`. Here should use RawReference.

Reply:
Maybe I should move this error-checking function into target so it's less error-prone.